src/neural-network.js

```diff
@@ -119,12 +119,42 @@ export default class NeuralNetwork {
     }
   }
 
+  /**
+   *
+   * @returns boolean
+   */
+  get isRunnable(){
+    if(!this.runInput){
+      console.error('Activation function has not been initialized, did you run train()?');
+      return false;
+    }
+
+    const checkFns = [
+      'sizes',
+      'outputLayer',
+      'biases',
+      'weights',
+      'outputs',
+      'deltas',
+      'changes',
+      'errors',
+    ].map(c => this[c]).every(fn => fn !== null);
+
+    if(!checkFns){
+      console.error('Some settings have not been initialized correctly, did you run train()?');
```

Comment:
which you could then use here, for a more specific error

Reply:
I like that much better. Check out the new implementation